### DISCUSSION TITLE
[Modding] How do I make a custom character

### DISCUSSION THREAD

**Discussion by Kristopian** [diasurgical/DevilutionX#3665] [Q&A] [answered]
Hi, I'm wondering how I would go about making a custom character with unique abilities, audio, and items.

**Comment by AJenbo** [comment]
You need to know a bit of C++ to be able to do so. 

The heros are defined in player.cpp/h

All the sounds are defined in effects.cpp/h

Abilities are spread all over the code, seach for `Warrior` to find most of them.

Item data is defined in itemdat.cpp

**Reply by Kristopian** [reply]
I don't know much C++ but I know enough to make a simple turn based game, and I know how to do simple top down games, and that's about all my C++ knowledge.

**Reply by Kristopian** [reply]
Sorry I did forget to mention I plan on making my mod a separate thing you can select like hellfire/freeware.

**Comment by Kristopian** [comment]
while working on one of my characters (the architect) I have run into an issue with his unique ability to make walls and then destroy those walls, and the solution I have come up with for this wall mechanic is to use minions that can't move or deal damage, but even with zero speed and zero attack it still slowly inches toward enemies and does a little damage, it should be a still object whom monsters can destroy and the architect can destroy using his second starting spell.

Edit-
How valid is making and rigging a character in blender and and then editing it in Photoshop

**Reply by julealgon** [reply]
> and the solution I have come up with for this wall mechanic is to use minions that can't move or deal damage, but even with zero speed and zero attack it still slowly inches toward enemies and does a little damage, it should be a still object

You realize it is a bit hard to provide advice since what you are doing is basically hacking existing systems to do stuff they were not designed to do. I'd encourage you to move away from these types of solutions.

**Reply by Kristopian** [reply]
I will try to find a different solution

**Reply by Kristopian** [reply]
> 
I have made a new layer for the walls and changed path-finding on several enemies (not all of them yet), they will check if walking around walls or breaking them down is faster and proceed towards their target, be it minions or me.

**Comment by Kristopian** [comment, marked as answer]
Gonna close this.
